ui/src/app/components/workflows/workflows-home/workflows-home.component.ts

```diff
@@ -63,8 +65,17 @@ export class WorkflowsHomeComponent implements OnInit, OnDestroy {
       });
   }
 
+  runWorkflow(id: number) {
+    this.runWorkflowDialogSubscription = this.confirmationDialogService
+      .confirm(ConfirmationDialogTypes.YesOrNo, texts.RUN_WORKFLOW_CONFIRMATION_TITLE, texts.RUN_WORKFLOW_CONFIRMATION_CONTENT)
+      .subscribe((confirmed) => {
+        if (confirmed) this.store.dispatch(new RunWorkflow(id));
+      });
+  }
+
   ngOnDestroy(): void {
     !!this.workflowsSubscription && this.workflowsSubscription.unsubscribe();
     !!this.confirmationDialogServiceSubscription && this.confirmationDialogServiceSubscription.unsubscribe();
+    !!this.runWorkflowDialogSubscription ? this.runWorkflowDialogSubscription.unsubscribe() : '';
```

Comment:
I think this version looks better !!this. runWorkflowDialogSubscription && !!this. runWorkflowDialogSubscription. unsubscribe()